### DISCUSSION TITLE
What about a score-oam?

### DISCUSSION THREAD

**Discussion by vitorfhc** [score-spec/spec#44] [Ideas]
I understood that Score abstracts some configurations for more complex layers, such as Docker and Helm.

After reading issue #23, I think Score is another layer of abstraction that could also be on top of OAM, which means we could develop a `score-oam`. Even though I love OAM, it doesn't decrease the cognitive load on developers as much as Score.

Please let me know what your thoughts are.

**Comment by sujaya-sys** [comment]
Interesting point! It's definitely something we'd have to explore in more detail to understand whether it makes sense in practice. My thoughts on this: 

- Which output would a `score-oam` integration produce? OAM/Kubevela consumes components such as Helm charts or Kustomize packages directly, meaning there's no standardised output format that you could translate the properties defined in your `score.yaml` file into. (I'm referring to the "Components" section [here](https://kubevela.io/docs/getting-started/core-concept)).

- Another idea is to work with target platform CLI's such as `score-helm` or `score-kustomize` directly and feed their outputs into the `components` and `traits` section of the OAM/kubevela file. 

As mentioned above, the main question for me is whether it would make sense for OAM users in practice. OAM - just as Score - advocates for a separation of concerns between developer and platform engineers. The Kubevela file is designed for and catered towards the developer persona as defined by OAM. Adding Score into the mix - which slices dev and platform responsibilities in a different way - would force developers to work with an additional file / tool and potentially complicate their workflow. 

Curious to hear your thoughts on this as well! 

I'd also like to pull in @fernando-villalba who recently published an [article](https://score.dev/blog/score-vs-open-application-model-kubevela) on this topic :-)

**Reply by stevehipwell** [reply]
@sujaya-sys I don't think you should be treating OAM and KubeVela as a single thing just because KubeVela is currently the only OSS system consuming OAM.

**Reply by yaronpri** [reply]
Hi @sujaya-sys  what about score-kustomize is something that is planned?

**Comment by stevehipwell** [comment]
@vitorfhc I think developer tooling built on top of OAM is what's needed in this space, as well as an alternative OAM implementation to KubeVela (I'd like to see a CLI tool and a simple operator). What I'm not sure about is if Score offers much of a benefit in this regard over building something separate?
